### PR TITLE
Update Windows Broker documentation to clarify MSA behaviours

### LINKED ISSUE #1355: Microsoft Account user model on Windows.
Hi @vtbassmatt , @wolf99 , @mjcheetham , I would like to learn more about what's the reason you mentioned this notes in the [windows-broker.md](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/windows-broker.md#for-microsoft-accounts):
 

> Two very important things to note:
>   
>   If you haven't connected any Microsoft accounts to Windows before, the first account you connect will cause the local Windows user account to be converted to a connected account.
>   In addition, you can't change the usage preference for the first Microsoft account connected to Windows: all Microsoft apps will be able to sign you in with that account.
>   As far as we can tell, there are no workarounds for either of these behaviors (other than to not use the WAM broker).

I believe there are some misunderstandings about how MSA works in WAM as well as SSO.
By adding the MSA via application such as GCM, the MSA can be added as associated (SSO) or secondary (CSO, one click sign on) and it's controlled by user.
In addition, it should not convert your local account into MSA. 

Please let me know what's your experience and the reason behind those notes that you have added.
Thanks.

### PR DESCRIPTION
**Pull request by mjcheetham**
When you sign-in with an MSA using WAM you are asked if you wish to use this account "everywhere on your device", or just "Microsoft apps only". The former will connect your local user account to an MSA, and the latter will not. Clarify the docs around this, including an annotated image of the
specific dialog.

In addition, since GCM 2.1, we are using a newer version of MSAL that uses the new MSAL runtime that solves the COM security issues that presented with elevated processes. Update our documentation to recommend people update to fix this problem.

Fixes #1355

Rendered: https://github.com/mjcheetham/git-credential-manager/blob/wam-doc-update/docs/windows-broker.md

### CODE REVIEW

File: docs/windows-broker.md [hunk 1]
```diff
@@ -181,8 +205,10 @@ See https://aka.ms/gcm/wamadmin for more information.
 
 In order to fix the problem, there are a few options:
 
-1. Run Git or Git Credential Manager from non-elevated processes.
-2. Disable the broker by setting the
+1. Update to the [latest Git for Windows][git-for-windows-latest]
+   **(recommended)**.
+2. Run Git or Git Credential Manager from non-elevated processes.
```
Comment (ldennington) [inline docs/windows-broker.md:210]:
```suggestion
2. Run Git or Git Credential Manager from a non-elevated processes.
```

**Review by ldennington — APPROVED** [review]
Added a few nitpicky suggestions - feel free to take them or leave them!

File: docs/windows-broker.md [hunk 2]
```diff
@@ -132,27 +135,48 @@ improve over time and a "personal account" option to be presented in the future.
...
-whether or not the account is available to other Microsoft applications in
-**Settings**, **Accounts**, **Emails & accounts**:
+authentication prompts when using GCM.
+
+---
+
+⚠️ **Important** ⚠️
+
+When adding a new MSA to Windows, you'll be asked to select whether to use this
+account across all of your device (**option 1**), or only permit Microsoft-apps
```
[the remark this answers is not in the document]
Comment (mjcheetham) [reply docs/windows-broker.md:145]: The text here is quoting from the Windows dialog. "across all of your device" (singular) is correct. This action only impacts one device.

File: docs/windows-broker.md [hunk 3]
```diff
@@ -132,27 +135,48 @@ improve over time and a "personal account" option to be presented in the future.
...
+to access your identity (**option 2**). If you opt to use the account everywhere
+then your local Windows user account will be connected to that MSA account.
+This means you'll need to use your MSA credentials to sign-in to Windows going
+forwards.
+
+Selecting "just this app" or "Microsoft apps only" will still allow you to use
+this MSA across apps in Windows, but will not require you to use your MSA
+credentials to sign-in to Windows.
+
+![Confirmation to connect your MSA to Windows][msa-confirm]
```
[the remark this answers is not in the document]
Comment (mjcheetham) [reply docs/windows-broker.md:155]: MSA stands for Microsoft Account, so MSA Account is "Microsoft Account Account" :-)

File: docs/windows-broker.md (hunk 1 above)
Comment (mjcheetham) [reply docs/windows-broker.md:210]: I'm using the plural "processes" here, so "a" isn't correct grammar.

File: docs/windows-broker.md [hunk 4]
```diff
@@ -132,27 +135,48 @@ improve over time and a "personal account" option to be presented in the future.
...
+authentication prompts when using GCM.
+
+---
+
+⚠️ **Important** ⚠️
+
+When adding a new MSA to Windows, you'll be asked to select whether to use this
+account across all of your device (**option 1**), or only permit Microsoft-apps
+to access your identity (**option 2**). If you opt to use the account everywhere,
+then your local Windows user account will be connected to that MSA account.
```
Comment (ldennington) [inline docs/windows-broker.md:147]: Perhaps we should just make this:

```suggestion
then your local Windows user account will be connected to that MSA.
```

**Review by ldennington — APPROVED** [review]
Added one last correction to change an "MSA account" to "MSA."

**Pull request merged**